src/Cli/dotnet/ToolPackage/IToolPackageDownloader.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Threading.Tasks;
+using Microsoft.DotNet.ToolPackage;
+using NuGet.Versioning;
+
+namespace Microsoft.DotNet.Cli.ToolPackage
+{
+    internal interface IToolPackageDownloader
+    {
+        IToolPackage InstallPackageAsync(PackageLocation packageLocation, PackageId packageId,
```

Comment:
If this has `Async` at the end of its name, then it should return `Task<IToolPackage>`, and the implementation should use `await` instead of `.GetAwaiter().Result`.  Then synchronous methods that call it would use `.GetAwaiter().GetResult()`.

Alternatively, you could change the name to `InstallPackage`.